### ISSUE TITLE
[Feature Request] partial zone editing

### ISSUE THREAD

**Issue by bryanpedini** [PowerDNS-Admin/PowerDNS-Admin#1220]
If it is not present yet, in case of which I would like to know how to configure it, I would like to propose the addition of partial zone editing for certain users.

First off, I don't know how it would be implemented, meaning I see no easy way of actually doing it code-wise, tho I'll explain it better:

I have a the reverse domain `0.1.168.192.in-addr.arpa` (for the subnet `192.168.1.0/24`) and would like to allow one user account to edit only the records `64` through `71` (for the subnet `192.168.1.64/29`) and another customer account to edit only the records `72` through `79` and so on.

We would need so to avoid the need of another custom-made interface to connect to the PowerDNS APIs to allow customers to edit their IP subnets' reverse records on their own instead of opening a ticket to us.

Thanks in advance.

**Comment by jbe-dw** [comment]
Hello,

There is no such feature, but I think I had a talk with a contributor who has the same needs. The idea would be to add permissions arround regex to edit zone subentries.

This won't be in our priorities however it may land by an external contribution.

Regards

**Comment by bryanpedini** [comment]
> A common solution to the shared reverse zone problem is to use `CNAME`s to move the `PTR` records into the forward zone, where the user already has write access.

whistle whistle...
I still think that the functionality requested would be beneficial and allow customers to have "less hassle" sort of speak, but man I didn't think of this one out...

thank you a lot, I'mma go experiment this right now!!
it may at least "fix" the thing for the time being

**Comment by jbe-dw** [comment]
You're right, there are few "hassling" stuff in permissions that needs improvement. But today we just manage zone level permissions, and it needs a huge lift to get to records. We use the "account" field of PowerDNS for this, but it is limiting for PDA permission management, as we don't store records locally.

**Comment by AzorianSolutions** [comment]
First off, thank you @corubba for that insight! I never knew about a work-around trick like the method you described. That's quite nifty.

@bryanpedini I can't make any promises as to whether this generation of the project will see such changes but I can tell you that I am specifically addressing this permissions issue in the next-generation project. I will use something like AWS's ARN scheme to provide a sort of structured reference to anything and everything. From there, it should make implementing ACL matching rules rather easy. For example:

Zone: example1.com
- @ A: arn:tenants:52554:zones:874432:records:908543:crud

ACL Rule: arn:tenants:52554:zones:874432:records:908*:ru

So the idea would be that with an easy GUI, this permissions system should accommodate a very high level of granularity without a lot of work. By all means, please provide your thoughts as the time is now while I'm in the planning / design phase.
